generator/src/test/java/com/scottlogic/deg/generator/cucumber/steps/GeneralTestStep.java

```diff
@@ -51,6 +54,11 @@ public void setTheCombinationStrategy(GenerationConfig.CombinationStrategyType s
         this.state.combinationStrategyType = strategy;
     }
 
+    @When("we do not violate any {operator} constraints")
+    public void constraintTypeIsNotViolated(String operator){
+        this.state.addConstraintToNotViolate(Arrays.stream(AtomicConstraintType.values()).filter(x->x.toString().equals(operator)).findFirst().get());
```

Comment:
Can you do `AtomicConstraintType.valueOf(operator)` here?

Reply:
doesn't work.

I think valueOf is a default method that checks for a string with the same text as the enum, rather than vs our text value.

I could override the valueOf method to check against text.

Reply:
Ok, how does it work elsewhere with cucumber, e.g. in `StringValueStep`

Reply:
You should be able to use `AtomicConstraintType.valueOf(state.extractConstraint(operator))` this should work if you make it package private.

Reply:
Doesn't work for the same reason. valueOf expects "ISLESSTHANCONSTANT", and we're passing around "lessThan"

Reply:
StringValueStep uses constraintDTO and does something else entirely with it